snapcraft/tests/fixture_setup.py

```diff
@@ -443,6 +444,47 @@ def _setUp(self):
         self.addCleanup(patcher.stop)
 
 
+class FakeSnapd(fixtures.Fixture):
+    '''...'''
+
+    def __init__(self, *, socket_error=False, api_error=False):
+        self._socket_error = socket_error
+        self._api_error = api_error
+
+    def _setUp(self):
+        patcher = mock.patch('requests_unixsocket.Session.request')
+        self.session_request_mock = patcher.start()
+        self.session_request_mock.side_effect = self.request_side_effect()
+        self.addCleanup(patcher.stop)
+
+    def request_side_effect(self):
+        def request_effect(*args, **kwargs):
+            if args[0] == 'GET' and '/v2/snaps/' in args[1]:
+                if self._socket_error:
+                    raise ConnectionError(
+                        'Connection aborted.',
+                        FileNotFoundError(2, 'No such file or directory'))
+
+                class Session:
+                    def __init__(self, name, api_error):
+                        self._name = name
+                        self._confinement = 'core' in name or 'classic'
```

Comment:
can you enlighten me what this logic implies or what `self._confinement` drives?